docs/sources/flow/reference/components/prometheus.exporter.postgres.md

```diff
@@ -0,0 +1,125 @@
+---
+# NOTE(rfratto): the title below has zero-width spaces injected into it to
+# prevent it from overflowing the sidebar on the rendered site. Be careful when
+# modifying this section to retain the spaces.
+#
+# Ideally, in the future, we can fix the overflow issue with css rather than
+# injecting special characters.
+
+title: prometheus.exporter.postgres
+labels:
+  stage: beta
+---
+
+# prometheus.exporter.postgres
+The `prometheus.exporter.postgres` component embeds
+[postgres_exporter](https://github.com/prometheus-community/postgres_exporter) for collecting metrics from a postgres database.
+
+Multiple `prometheus.exporter.postgres` components can be specified by giving them different 
+labels.
+
+## Usage
+
+```river
+prometheus.exporter.postgres "LABEL" {
+    data_source_names = DATA_SOURCE_NAMES_LIST
+}
+```
+
+## Arguments
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`data_source_names`                  | `list(secret)`      | Specifies the Postgres server(s) to connect to.  |         | yes
+`disable_settings_metrics`           | `bool`              | Disables collection of metrics from pg_settings. | `false` | no
+`disable_default_metrics`            | `bool`              | When `true`, only exposes metrics supplied from `custom_queries_path`. | `false` | no
+`autodiscover_databases`             | `bool`              | Whether to automatically discover databases to scrape metrics from. | `false` | no
+`exclude_databases`                  | `list(string)`      | A list of databases to ignore when `autodiscover_databases` is `true` | | no
+`include_databases`                  | `list(string)`      | Includes only specific databases (excluding all others) when autodiscover_databases is `true` | | no
```

Comment:
This interaction between fields is confusing to me. Even after reading it a few times. Is it ever valid to have auto: true and an include list?

Reply:
I've changed the structure a bit to hopefully make it more clear that those config fields are linked together, as well as updated the documentation with more details and examples.